sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/EventLogger.kt

```diff
@@ -225,13 +230,21 @@ object EventLogger {
                     eventsStorage.updateEvent(eventId, event.setType(EventType.WARNING.displayName))
                 }
             )
-            // do nothing
-            else -> {}
+            else -> { /* do nothing */ }
         }
     }
 
-    private fun registerToAppTransitions() {
-        // ToDo
+    private fun registerToAppTransitions(observer: AppLifecycleObserver) {
+        observer.registerListener(object : LifecycleListener {
+            // When app transitioned to foreground, send all events if TTL expired
+            override fun becameForeground() {
+                executorService.safeExecute {
+                    if (isTtlExpired()) {
+                        sendAllEventsInStorage()
```

Comment:
Is there any chances that **sendAllEventsInStorage** might be called 2 times consecutively?

Reply:
Current TTL value is 12 hours, so it is unlikely for `sendAllEventsInStorage` be called almost consecutively. Also `sendAllEventsInStorage` is safe to call multiple times, since it will check if empty, then do nothing if empty